rllib/algorithms/cql/cql.py

```diff
@@ -122,6 +137,38 @@ def training(
 
         return self
 
+    @override(SACConfig)
+    def get_default_learner_class(self) -> Union[Type["Learner"], str]:
+        if self.framework_str == "torch":
+            from ray.rllib.algorithms.cql.torch.cql_torch_learner import CQLTorchLearner
+
+            return CQLTorchLearner
+        else:
+            raise ValueError(
+                f"The framework {self.framework_str} is not supported. "
+                "Use `'torch'` instead."
+            )
+
+    @override(AlgorithmConfig)
+    def build_learner_connector(
+        self,
```

Comment:
see my comment in CQLLearner. I think we should move this there. Same as for DQN. Or if there  are good arguments to leave these in the Config classes, we should also change it in DQN/SAC.

Reply:
As mentioned above. In rare cases the class inheritance avoids it being changed there and changing it in `build_learner_connector` would then still work.

You are right that we need to unify the way of adding them in all algorithms (also MARWIL)

Reply:
It just doesn't feel clean, doing it here. The config classes should contain as little (actually implemented) algo logic as possible and just store settings.

I know I sound like a broken record, but "separation of concerns" principle :):
* Whatever an algo-specific Learner must-have, it should create this in its `build()` method, for example a lr-schedule, or a kl-coeff-schedule, etc..
* Whatever an algo-specific Learner must-have, it should check that it has this in its `build()` method, e.g. "does my RLModule implement API xyz?"
* Algo configs should NOT implement any algo logic. If they still do somewhere for some algos, we should add a TODO to move the logic into a more appropriate location.

Suggestion:
Can we try creating a `CQLLearner` class that inherits from `SACLearner` and overrides the `build()` method just to insert that connector piece? Then do `class CQLTorchLearner(SACTorchLearner, CQLLearner):`. That should work, no?